src/sql/workbench/browser/modelComponents/inputbox.component.ts

```diff
@@ -245,6 +245,11 @@ export default class InputBoxComponent extends ComponentBase<azdata.InputBoxProp
 
 		input.inputElement.required = this.required;
 		input.inputElement.readOnly = this.readOnly;
+
+		// only update title if there's a value, otherwise title gets set to placeholder above
+		if (this.title) {
```

Comment:
do an explicit check with undefined, if someone has for some reason a value of '0' for the title then that won't get set and which is going to surprise the user.

Reply:
That's only true for the number 0 - title is a string and thus the string '0' is not falsy and so will pass this check 

https://developer.mozilla.org/en-US/docs/Glossary/Falsy

Reply:
Charles is right! I was confused :)

Reply:
it gets set correctly to 0 if it's set to the string '0'. Here's an example of where title was set to '0' for the input box where the placeholder is "Select connection"
![image](https://user-images.githubusercontent.com/31145923/97236496-73362080-17a2-11eb-8edd-407539b122fd.png)

Is there a different case you were thinking of?